cdc/sink/codec/json.go

```diff
@@ -615,12 +615,14 @@ func (d *JSONEventBatchEncoder) Reset() {
 func (d *JSONEventBatchEncoder) SetParams(params map[string]string) error {
 	var err error
 
-	d.maxMessageBytes = config.DefaultMaxMessageBytes
-	if maxMessageBytes, ok := params["max-message-bytes"]; ok {
-		d.maxMessageBytes, err = strconv.Atoi(maxMessageBytes)
-		if err != nil {
-			return cerror.ErrSinkInvalidConfig.Wrap(err)
-		}
```

Comment:
The original logic is not elegant, but what is the bug in it?
- If `max-message-bytes` is set in sink-uri, the following lines set it: https://github.com/pingcap/tiflow/blob/43a599d08044606dc9af8149a79f92d2cc45d08f/cdc/sink/mq.go#L419-L422
- If `max-message-bytes` is not set in sink-uri, `config.DefaultMaxMessageBytes` will be used here.

Reply:
we have to update the config once more.because we would query Kafka broker to get `topicMaxMessageBytes` or `brokerMessageMaxBytes`, and then choose the minimum of min(`max-message-bytes`, `topicMaxMessageBytes`, `brokerMessageMaxBytes`)

Reply:
make sense, maybe more scenario tests can be added, either in integration test or test infra.

Reply:
we have to refact the configuration in the kafka, `Config`, `sarama.Config`, `opts`, and cover the adjust logic.